### PR TITLE
fix: support db passed to DBRef constructor (COMPASS-4835)

### PR DESCRIPTION
**Pull request by lerouxb**
**Related:**

* https://github.com/mongodb-js/compass/pull/2420

See discussion here https://mongodb.slack.com/archives/C0V8RU15L/p1629898147069600

See also https://github.com/mongodb/js-bson/blob/master/src/db_ref.ts

### CODE REVIEW

**Comment by lerouxb** [comment]
```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/mongodb-js/pre-commit.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '140.82.114.3' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-08-25T14_49_19_792Z-debug.log
Error: Process completed with exit code 1.
```

**Review by addaleax — APPROVED** [review]
Do we want to go one step further and also account for the fact that db and collection names may include quotes?

Overall, this is very similar to https://github.com/mongodb-js/mongosh/blob/main/packages/service-provider-core/src/printable-bson.ts, maybe we can eventually avoid having a separate stringification package here

(And while we’re at it, the `Timestamp()` argument order should probably be reversed – `Timestamp(high, low)` is the shell-accepted signature and `Timestamp(low, high)` is deprecated in the Node.js driver)

**Comment by lerouxb** [comment]
I'll fix Timestamp now. Adding tests..

File: package-lock.json [hunk 1]
```diff
@@ -2347,9 +6188,9 @@
       }
     },
     "pre-commit": {
-      "version": "github:mongodb-js/pre-commit#f4158768a7ef58b46808cc09a6f6a0994c0baa67",
-      "from": "github:mongodb-js/pre-commit",
+      "version": "git+ssh://git@github.com/mongodb-js/pre-commit.git#f4158768a7ef58b46808cc09a6f6a0994c0baa67",
       "dev": true,
+      "from": "pre-commit@github:mongodb-js/pre-commit",
```
Comment (gribnoysup) [inline package-lock.json:6193]: This is probably why the CI is failing

Comment (lerouxb) [reply package-lock.json:6193]: Any objections if I just remove the pre-commit hook?

**Comment by lerouxb** [comment]
@addaleax Actually I don't really want to wade into Timestamp right now: https://github.com/mongodb/js-bson/blob/master/src/timestamp.ts#L31-L42

None of those signatures seem to be high then low.

**Pull request merged**